intake_esgf/core/globus.py

```diff
@@ -64,6 +64,7 @@ def search(self, **search: Union[str, list[str]]) -> pd.DataFrame:
 
         # build up the query and search
         query_data = SearchQuery("")
+        search = {k: v for k, v in search.items() if len(v) > 0}
```

Comment:
Ah, some search facets are of type `bool` and do not have a `len`. Perhaps we need a `isinstance(v,str)` also? Ultimately I think we need to move this to `ESGFCatalog.search()`. This way it will get logged that we dropped this facet that you searched.

Reply:
Added a check for str facet and moved prior to logging